src/App.tsx

```diff
@@ -1,14 +1,54 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
+import { Todo } from './types/Todo';
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { getTodos } from './api';
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoadEnd, setIsLoadEnd] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo>();
+  const [filter, setFilter] = useState('all');
+  const [searchText, setSearchText] = useState('');
+
+  useEffect(() => {
+    getTodos().then(todo => setTodos(todo))
+      .finally(() => setIsLoadEnd(true));
+  }, []);
+
+  const sortedTodo = todos.filter(currTodo => {
+    switch (filter) {
+      case 'all':
+        if (currTodo.title.includes(searchText)) {
+          return true;
+        }
+
+        break;
+      case 'active':
+        if (currTodo.completed === false
+          && currTodo.title.includes(searchText)) {
+          return true;
+        }
+
+        break;
+      case 'completed':
+      default:
+        if (currTodo.completed === true
+          && currTodo.title.includes(searchText)) {
+          return true;
+        }
+    }
+
+    return false;
+  });
```

Comment:
```suggestion
const sortedTodo = () => {
    switch (selectBy) {
      case'all':
        return todos;
      case 'active':
        return todos.filter(todo => !todo.completed);
      case 'completed':
      default:
        return todos.filter(todo => todo.completed);
    }
  };
```
This code will work faster because your filter will not start in case of None or the default, on other hand in your solution it will look at each element and do nothing...